src/LibraryManager.Contracts/FileHelpers.cs

```diff
@@ -17,6 +17,42 @@ namespace Microsoft.Web.LibraryManager.Contracts
     /// </summary>
     public static class FileHelpers
     {
+        /// <summary>
+        /// Writes the stream to a temporary file first, then moves the temporary file to the destination file
+        /// </summary>
+        /// <param name="fileName"></param>
+        /// <param name="sourceStream"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public static async Task<bool> SafeWriteToFileAsync(string fileName, Stream sourceStream, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            string tempFileName = Path.Combine(
+                Environment.GetFolderPath(Environment.SpecialFolder.InternetCache),
+                "libman",
+                fileName.GetHashCode().ToString(),
+                Path.GetFileName(fileName));
```

Comment:
I wonder if we should just use Path.GetTempFileName for now, until we nave a need for something more complicated?

Reply:
Haha, OK, I just commented on that above. OK, I'll change that.

Reply:
Yep, ended up just doing that.